unic/ucd/normal/src/canonical_combining_class.rs

```diff
@@ -12,44 +12,98 @@
 
 //! Accessor for Canonical_Combining_Class (ccc) property
 
-// TODO: [API] Support Canonical_Combining_Class_Values, either by creating a separate `enum` type,
-// or converting `CanonicalCombiningClass` into enum and allowing access to numerical value.
-// <http://unicode.org/reports/tr44/#Canonical_Combining_Class_Values>
-
-// TODO: Use associate type `u8`
 
 use std::cmp::Ordering;
 
 
 /// Represents *Canonical_Combining_Class* property of a Unicode character.
 ///
 /// * <http://unicode.org/reports/tr44/#Canonical_Combining_Class>
-pub trait CanonicalCombiningClass {
-    /// If the *ccc* has value `Not_Reordered` (`0`).
-    fn ccc_is_not_reordered(&self) -> bool;
+#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct CanonicalCombiningClass(u8);
+
+
+// TODO: Once we fully adopt 1.20 change these to associated consts on CanonicalCombiningClass
+/// Canonical Combining Classes by their name
+#[allow(non_upper_case_globals)]
+pub mod values {
```

Comment:
Based on the previous discussion, should this be `long_names`, as it's aliasing Long Names for Property Values?

And, we can also add the `pub mod abbr_names`, based on <http://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt>, if you like to make it more complete.